kimtaeheon/likelion-assignment-crud/src/main/java/com/likelion/likelioncrud/tag/application/TagService.java

```diff
@@ -0,0 +1,67 @@
+package com.likelion.likelioncrud.tag.application;
+
+import com.likelion.likelioncrud.common.error.ErrorCode;
+import com.likelion.likelioncrud.common.exception.BusinessException;
+import com.likelion.likelioncrud.tag.api.dto.request.TagSaveRequestDto;
+import com.likelion.likelioncrud.tag.api.dto.request.TagUpdateRequestDto;
+import com.likelion.likelioncrud.tag.api.dto.response.TagInfoResponseDto;
+import com.likelion.likelioncrud.tag.api.dto.response.TagListResponseDto;
+import com.likelion.likelioncrud.tag.domain.Tag;
+import com.likelion.likelioncrud.tag.domain.repository.TagRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Transactional(readOnly = true)
+@RequiredArgsConstructor
+public class TagService {
+
+    private final TagRepository tagRepository;
+
+    @Transactional
+    public void tagSave(TagSaveRequestDto tagSaveRequestDto) {
+        Optional<Tag> existingTag = tagRepository.findByName(tagSaveRequestDto.name());
+        if (existingTag.isPresent()) {
+            return;
+        }
```

Comment:
여기서 중복 저장 안 되는 로직 추가해 주신 것 같은데 중복 이름으로 저장시켜보면 요청은 들어가서 200 ok이지만 데베에는 저장되지 않네요.
이때 return; 보다는 예외처리 메세지를 주는 건 어떨까요 ??

```suggestion
        if (existingTag.isPresent()) {
        throw new BusinessException(ErrorCode.TAG_ALREADY_EXISTS, ErrorCode.TAG_ALREADY_EXISTS.getMessage());
        }
```
위와 같이 예외를 던져준다면 400 BadRequest로 태그의 중복 저장을 막고 사용자에게 예외 메세지를 줌으로써 적절한 피드백 메세지를 전달해 줄 수 있겠죠 ?! 🙂🙂

Reply:
넵!! 예외처리 했습니다!